datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/AbstractKafkaBasedConnectorTask.java

```diff
@@ -229,8 +232,13 @@ protected void translateAndSendBatch(ConsumerRecords<?, ?> records, Instant read
     for (TopicPartition topicPartition : records.partitions()) {
       for (ConsumerRecord<?, ?> record : records.records(topicPartition)) {
         try {
-          if (_autoPausedSourcePartitions.containsKey(topicPartition)) {
-            _logger.warn("Abort sending as {} is auto-paused, rewind offset", topicPartition);
+          boolean skipRecord;
+          synchronized (_sendFailureTopicPartitionExceptionMap) {
+            skipRecord = _autoPausedSourcePartitions.containsKey(topicPartition)
+                || _sendFailureTopicPartitionExceptionMap.containsKey(topicPartition);
+          }
+          if (skipRecord) {
+            _logger.warn("Abort sending as {} is auto-paused or saw a send failure, rewind offset", topicPartition);
```

Comment:
Do you see a value in clarifying in the warning whether skipping the record because of autoPausedSourcePartition or send failure ?

Reply:
Sure, I can make that distinction